<div style=" background-color: RGB(0,114,200);" >
<h1 style="margin: auto; padding: 20px 0; color:#fff; text-align: center">PROJET OF DATA ANALYSIS</h1>
<h2 style="margin: auto; padding: 20px 0; color:#fff; text-align: center">FAO Study
</h2>
</div>

![FAO](FAO.png)


# RESEARCH QUESTION

* What are the global trends in undernutrition over the past decade, and which regions have been most affected?
* How factors such as import, disponibility... correlate with undernutrition rates in different countries?
* which demographic groups have been most vulnerable? which country import most, have most disponibility... in 2017





Source of dataset : https://www.kaggle.com/code/alizechasse/fao-study/notebook

<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Step 1 - Importation of librairies and loading data</h2>
</div>

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">1.1 - Importation of librairies</h3>
</div>

In [803]:
import pandas as pd
import numpy as np

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">1.2 - Load csv files</h3>
</div>

In [804]:
# population.csv
population = pd.read_csv('./data/population.csv', sep=',')

# disponibility.csv
disponibility = pd.read_csv('./data/disponibility.csv', sep=',')

# alimentary.csv
alimentary = pd.read_csv('./data/alimentary.csv', sep=',')

# nutrition.csv
nutrition = pd.read_csv('./data/nutrition.csv', sep=',')


<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Step 2 - EDA</h2>
</div>

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.1 - EDA of population</h3>
</div>

In [805]:
population.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1416 entries, 0 to 1415
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   country     1416 non-null   object 
 1   year        1416 non-null   int64  
 2   Population  1416 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 33.3+ KB


* no nan values
* 1416 rows, 3 columns
* population dtype is float which need to be changed

In [806]:
population.head()

country  year  Population
0  Afghanistan  2013   32269.589
1  Afghanistan  2014   33370.794
2  Afghanistan  2015   34413.603
3  Afghanistan  2016   35383.032
4  Afghanistan  2017   36296.113

In [807]:
# changing population to million with int dtype
population['Population'] = population['Population'] * 1000
population['Population'] = population['Population'].astype(int)

In [808]:
population.head()

country  year  Population
0  Afghanistan  2013    32269589
1  Afghanistan  2014    33370794
2  Afghanistan  2015    34413603
3  Afghanistan  2016    35383032
4  Afghanistan  2017    36296113

In [809]:
population.year.unique()

array([2013, 2014, 2015, 2016, 2017, 2018], dtype=int64)

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.2 - EDA disponibility</h3>
</div>

In [810]:
disponibility.head()

country                product    Origin  for animals  other use  \
0  Afghanistan       Abats Comestible   animale          NaN        NaN   
1  Afghanistan        Agrumes, Autres  vegetale          NaN        NaN   
2  Afghanistan  Aliments pour enfants  vegetale          NaN        NaN   
3  Afghanistan                 Ananas  vegetale          NaN        NaN   
4  Afghanistan                Bananes  vegetale          NaN        NaN   

   Dispo(Kcal/p/d)  Dispo(kg/p/y)  Dispo grasse(g/p/d)  Dispo protein (g/p/d)  \
0              5.0           1.72                 0.20                   0.77   
1              1.0           1.29                 0.01                   0.02   
2              1.0           0.06                 0.01                   0.03   
3              0.0           0.00                  NaN                    NaN   
4              4.0           2.70                 0.02                   0.05   

   Dispo in  Exportations  Importations  nutrition  lost  Production  \
0      53.0           NaN           NaN       53.0   NaN        53.0   
1      41.0           2.0          40.0       39.0   2.0         3.0   
2       2.0           NaN           2.0        2.0   NaN         NaN   
3       0.0           NaN           0.0        0.0   NaN         NaN   
4      82.0           NaN          82.0       82.0   NaN         NaN   

   Semences  Traitement  Variation  
0       NaN         NaN        NaN  
1       NaN         NaN        NaN  
2       NaN         NaN        NaN  
3       NaN         NaN        NaN  
4       NaN         NaN        NaN

In [811]:
disponibility.describe(include='all')

country           product    Origin    for animals      other use  \
count    15605             15605     15605    2720.000000    5496.000000   
unique     174                98         2            NaN            NaN   
top       Inde  Abats Comestible  vegetale            NaN            NaN   
freq        97               174     11896            NaN            NaN   
mean       NaN               NaN       NaN     479.501838     157.391376   
std        NaN               NaN       NaN    4240.119637    5076.785816   
min        NaN               NaN       NaN       0.000000       0.000000   
25%        NaN               NaN       NaN       0.000000       0.000000   
50%        NaN               NaN       NaN       4.000000       0.000000   
75%        NaN               NaN       NaN      74.000000       4.000000   
max        NaN               NaN       NaN  150000.000000  347309.000000   

        Dispo(Kcal/p/d)  Dispo(kg/p/y)  Dispo grasse(g/p/d)  \
count      14241.000000   14015.000000         11794.000000   
unique              NaN            NaN                  NaN   
top                 NaN            NaN                  NaN   
freq                NaN            NaN                  NaN   
mean          34.789832       8.719368             1.283111   
std          107.287655      24.618223             3.680399   
min          -21.000000      -1.930000            -0.030000   
25%            0.000000       0.060000             0.010000   
50%            4.000000       0.830000             0.080000   
75%           21.000000       5.190000             0.630000   
max         1711.000000     430.760000            60.760000   

        Dispo protein (g/p/d)       Dispo in  Exportations  Importations  \
count            11561.000000   15382.000000  12226.000000  14852.000000   
unique                    NaN            NaN           NaN           NaN   
top                       NaN            NaN           NaN           NaN   
freq                      NaN            NaN           NaN           NaN   
mean                 1.223608     640.293460    110.596925     87.264543   
std                  3.598686    9067.267153   1053.318990    717.372714   
min                 -0.370000   -3430.000000    -41.000000   -201.000000   
25%                  0.010000       0.000000      0.000000      0.000000   
50%                  0.100000       7.000000      0.000000      2.000000   
75%                  0.660000      76.750000      9.000000     18.000000   
max                 54.970000  739267.000000  42797.000000  63381.000000   

            nutrition          lost     Production      Semences  \
count    14015.000000   4278.000000    9180.000000   2091.000000   
unique            NaN           NaN            NaN           NaN   
top               NaN           NaN            NaN           NaN   
freq              NaN           NaN            NaN           NaN   
mean       347.931359    106.053763    1090.379085     73.974653   
std       4475.704458   1113.100416   12067.344094    528.069224   
min       -246.000000      0.000000       0.000000      0.000000   
25%          0.000000      0.000000       2.000000      0.000000   
50%          5.000000      4.000000      22.000000      2.000000   
75%         52.000000     26.000000     191.250000     17.000000   
max     426850.000000  55047.000000  739267.000000  17060.000000   

           Traitement     Variation  
count     2292.000000   6776.000000  
unique            NaN           NaN  
top               NaN           NaN  
freq              NaN           NaN  
mean       961.905323    -15.407615  
std      10381.795904    549.834540  
min        -19.000000 -39863.000000  
25%          0.000000      0.000000  
50%          6.000000      0.000000  
75%         69.000000      0.000000  
max     326711.000000   5284.000000

In [812]:
disponibility.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15605 entries, 0 to 15604
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   country                15605 non-null  object 
 1   product                15605 non-null  object 
 2   Origin                 15605 non-null  object 
 3   for animals            2720 non-null   float64
 4   other use              5496 non-null   float64
 5   Dispo(Kcal/p/d)        14241 non-null  float64
 6   Dispo(kg/p/y)          14015 non-null  float64
 7   Dispo grasse(g/p/d)    11794 non-null  float64
 8   Dispo protein (g/p/d)  11561 non-null  float64
 9   Dispo in               15382 non-null  float64
 10  Exportations           12226 non-null  float64
 11  Importations           14852 non-null  float64
 12  nutrition              14015 non-null  float64
 13  lost                   4278 non-null   float64
 14  Production             9180 non-null   float64
 15  Se

* to add more insights i replace nan values by 0 to avoid deleting columns, imputation by other methos risky cuz we have a lot
of nan values

In [813]:
# replace NaN by 0
disponibility = disponibility.fillna(0)

In [814]:
disponibility.head()

country                product    Origin  for animals  other use  \
0  Afghanistan       Abats Comestible   animale          0.0        0.0   
1  Afghanistan        Agrumes, Autres  vegetale          0.0        0.0   
2  Afghanistan  Aliments pour enfants  vegetale          0.0        0.0   
3  Afghanistan                 Ananas  vegetale          0.0        0.0   
4  Afghanistan                Bananes  vegetale          0.0        0.0   

   Dispo(Kcal/p/d)  Dispo(kg/p/y)  Dispo grasse(g/p/d)  Dispo protein (g/p/d)  \
0              5.0           1.72                 0.20                   0.77   
1              1.0           1.29                 0.01                   0.02   
2              1.0           0.06                 0.01                   0.03   
3              0.0           0.00                 0.00                   0.00   
4              4.0           2.70                 0.02                   0.05   

   Dispo in  Exportations  Importations  nutrition  lost  Production  \
0      53.0           0.0           0.0       53.0   0.0        53.0   
1      41.0           2.0          40.0       39.0   2.0         3.0   
2       2.0           0.0           2.0        2.0   0.0         0.0   
3       0.0           0.0           0.0        0.0   0.0         0.0   
4      82.0           0.0          82.0       82.0   0.0         0.0   

   Semences  Traitement  Variation  
0       0.0         0.0        0.0  
1       0.0         0.0        0.0  
2       0.0         0.0        0.0  
3       0.0         0.0        0.0  
4       0.0         0.0        0.0

In [815]:
#multiplication de toutes les lignes contenant des milliers de tonnes en Kg


In [816]:
#Affichage les 5 premières lignes de la table


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.3 - EDA alimentary</h3>
</div>

In [817]:
alimentary.head()


country  year              product  value
0  Afghanistan  2013  Autres non-céréales    682
1  Afghanistan  2014  Autres non-céréales    335
2  Afghanistan  2013         Blé et Farin  39224
3  Afghanistan  2014         Blé et Farin  15160
4  Afghanistan  2013             Céréales  40504

In [818]:
alimentary.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1475 entries, 0 to 1474
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   country  1475 non-null   object
 1   year     1475 non-null   int64 
 2   product  1475 non-null   object
 3   value    1475 non-null   int64 
dtypes: int64(2), object(2)
memory usage: 46.2+ KB


* no nan values

In [819]:
alimentary['product'].unique()

array(['Autres non-céréales', 'Blé et Farin', 'Céréales',
       'Fruits secs, total', 'Huiles végétales',
       'Légumineuses Sèches,Tot.', 'Non-céréales', 'Riz, total',
       'Sucre, total', 'Céréales Secondaires', 'Lait, total',
       'Mélanges et préparations', 'Poiss&produi', 'BulgurFarEnt',
       'Viande Total', 'Graisses Com'], dtype=object)

* we have 22 type of alimentary

In [820]:
# multiply values from ton to kg (1000)
alimentary['value'] = alimentary['value'] * 1000


In [821]:
alimentary.head()


country  year              product     value
0  Afghanistan  2013  Autres non-céréales    682000
1  Afghanistan  2014  Autres non-céréales    335000
2  Afghanistan  2013         Blé et Farin  39224000
3  Afghanistan  2014         Blé et Farin  15160000
4  Afghanistan  2013             Céréales  40504000

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.3 - EDA nutrition</h3>
</div>

In [822]:
nutrition.head()


country       year p/nutrition
0  Afghanistan  2012-2014         8.6
1  Afghanistan  2013-2015         8.8
2  Afghanistan  2014-2016         8.9
3  Afghanistan  2015-2017         9.7
4  Afghanistan  2016-2018        10.5

In [823]:
nutrition.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1218 entries, 0 to 1217
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   country      1218 non-null   object
 1   year         1218 non-null   object
 2   p/nutrition  624 non-null    object
dtypes: object(3)
memory usage: 28.7+ KB


* change dtype of p/nutrition into int, year into int also chose mean year

In [824]:
nutrition.describe(include='all')

country       year p/nutrition
count          1218       1218         624
unique          203          6         139
top     Afghanistan  2012-2014        <0.1
freq              6        203         120

* we have <0.1, i decide to change it to 0 to avoid error while manupilate of data

In [825]:
nutrition['p/nutrition'] = nutrition['p/nutrition'].replace(['<0.1'], 0)

In [826]:
# change dtype
nutrition['p/nutrition'] = pd.to_numeric(nutrition['p/nutrition'])

In [827]:
# product personne who be nutrited to millions
nutrition['p/nutrition'] = nutrition['p/nutrition'] * 1e6

In [828]:
nutrition = nutrition.fillna(0)

In [829]:
nutrition['p/nutrition'] = nutrition['p/nutrition'].astype(int)

In [830]:
nutrition['year'] = nutrition['year'].replace(['2012-2014'], 2013)
nutrition['year'] = nutrition['year'].replace(['2013-2015'], 2014)
nutrition['year'] = nutrition['year'].replace(['2014-2016'], 2015)
nutrition['year'] = nutrition['year'].replace(['2015-2017'], 2016)
nutrition['year'] = nutrition['year'].replace(['2016-2018'], 2017)

In [831]:
nutrition.head()


country  year  p/nutrition
0  Afghanistan  2013      8600000
1  Afghanistan  2014      8800000
2  Afghanistan  2015      8900000
3  Afghanistan  2016      9700000
4  Afghanistan  2017     10500000

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.1 - Number, Proportion pop who are undernourished in 2017</h3>
</div>

In [832]:
pop_2017 = population.loc[population['year'] == 2017]
pop_2017 = pop_2017.copy()

In [833]:
nutrition_2017 = nutrition.loc[nutrition['year'] == 2017]
nutrition_2017 = nutrition_2017.copy()

In [834]:
# join df for 2017
nut_pop2017 = pd.merge(pop_2017, nutrition_2017, on=('year', 'country'))
nut_pop2017.drop('year',axis=1, inplace=True)

In [835]:
nut_pop2017.head()

country  Population  p/nutrition
0     Afghanistan    36296113     10500000
1  Afrique du Sud    57009756      3100000
2         Albanie     2884169       100000
3         Algérie    41389189      1300000
4       Allemagne    82658409            0

In [836]:
total_pop_nut = nut_pop2017.Population.sum()

In [837]:
print("the number of total population who be nutrited {}".format((total_pop_nut/1e9).round(2)), 'millard')

the number of total population who be nutrited 7.54 millard


In [838]:
prop_pop_nut =  nut_pop2017['p/nutrition'].sum()/ total_pop_nut * 100
print("the propotion of people who be nutrited in the word on 2017 is : {}".format((prop_pop_nut).round(2)), '%')

the propotion of people who be nutrited in the word on 2017 is : 7.1 %


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.2 -Theoretical number of people who could be fed in 2017</h3>
</div>

* how many calories take by person is 2000 source : https://www.healthhub.sg/live-healthy/how-much-to-eat-at-each-meal

In [839]:
# we take disponability by kcal, kg by person
dispo_person_2017 = disponibility[['country','Dispo(Kcal/p/d)', 'Dispo(kg/p/y)']]
dispo_person_2017 = dispo_person_2017.copy()
dispo_person_2017['Dispo(Kcal/p/y)'] = dispo_person_2017['Dispo(Kcal/p/d)'] * 365 # day to year
dispo_person_2017.drop('Dispo(Kcal/p/d)',axis=1, inplace=True)
dispo_person_2017.head()

country  Dispo(kg/p/y)  Dispo(Kcal/p/y)
0  Afghanistan           1.72           1825.0
1  Afghanistan           1.29            365.0
2  Afghanistan           0.06            365.0
3  Afghanistan           0.00              0.0
4  Afghanistan           2.70           1460.0

In [840]:
dispo_alim_country = dispo_person_2017[['country','Dispo(Kcal/p/y)']]
dispo_alim_country = dispo_alim_country.groupby(['country']).sum()



In [841]:
dispo_kcal_pop17 = pd.merge(dispo_alim_country, pop_2017, on = ('country'))
dispo_kcal_pop17 = dispo_kcal_pop17.drop(['country'], axis=1)
dispo_kcal_pop17.head()

Dispo(Kcal/p/y)  year  Population
0         761755.0  2017    36296113
1        1102300.0  2017    57009756
2        1163620.0  2017     2884169
3        1201945.0  2017    41389189
4        1278595.0  2017    82658409

In [842]:
#Création de la colonne dispo_kcal avec calcul des kcal disponibles mondialement


In [843]:
#Calcul du nombre d'humains pouvant être nourris


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.3 - Nombre théorique de personne qui pourrait être nourrie avec les produits végétaux</h3>
</div>

In [844]:
#Transfert des données avec les végétaux dans un nouveau dataframe


In [845]:
#Calcul du nombre de kcal disponible pour les végétaux


In [846]:
#Calcul du nombre d'humains pouvant être nourris avec les végétaux


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.4 - Utilisation de la disponibilité intérieure</h3>
</div>

In [847]:
#Calcul de la disponibilité totale


In [848]:
#création d'une boucle for pour afficher les différentes valeurs en fonction des colonnes aliments pour animaux, pertes, nourritures, 


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.5 - Utilisation des céréales</h3>
</div>

In [849]:
#Création d'une liste avec toutes les variables


In [850]:
#Création d'un dataframe avec les informations uniquement pour ces céréales


In [851]:
#Affichage de la proportion d'alimentation animale


In [852]:
#Affichage de la proportion d'alimentation animale


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.6 - Pays avec la proportion de personnes sous-alimentée la plus forte en 2017</h3>
</div>

In [853]:
#Création de la colonne proportion par pays


In [854]:
#affichage après trie des 10 pires pays


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.7 - Pays qui ont le plus bénéficié d'aide alimentaire depuis 2013</h3>
</div>

In [855]:
#calcul du total de l'aide alimentaire par pays


In [856]:
#affichage après trie des 10 pays qui ont bénéficié le plus de l'aide alimentaire


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.8 - Evolution des 5 pays qui ont le plus bénéficiés de l'aide alimentaire entre 2013 et 2016</h3>
</div>

In [857]:
#Création d'un dataframe avec la zone, l'année et l'aide alimentaire puis groupby sur zone et année 


In [858]:
#Création d'une liste contenant les 5 pays qui ont le plus bénéficiées de l'aide alimentaire


In [859]:
#On filtre sur le dataframe avec notre liste


In [860]:
# Affichage des pays avec l'aide alimentaire par année


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.9 - Pays avec le moins de disponibilité par habitant</h3>
</div>

In [861]:
#Calcul de la disponibilité en kcal par personne par jour par pays


In [862]:
#Affichage des 10 pays qui ont le moins de dispo alimentaire par personne 


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.10 - Pays avec le plus de disponibilité par habitant</h3>
</div>

In [863]:
#Affichage des 10 pays qui ont le plus de dispo alimentaire par personne 


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.11 - Exemple de la Thaïlande pour le Manioc</h3>
</div>

In [864]:
#création d'un dataframe avec uniquement la Thaïlande 


In [865]:
#Calcul de la sous nutrition en Thaïlande


In [866]:
# On calcule la proportion exportée en fonction de la proportion


<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Etape 6 - Analyse complémentaires</h2>
</div>

In [867]:
#Rajouter en dessous toutes les analyses complémtaires suite à la demande de mélanie :
#"et toutes les infos que tu trouverais utiles pour mettre en relief les pays qui semblent être 
#le plus en difficulté au niveau alimentaire"